### ISSUE TITLE
Groupchats

### ISSUE THREAD

**Issue by subliun** [subliun/Antox#103]

**Comment by zoff99** [comment]
the text "coming soon" is been ther for a long time in antox.
whats going on? any progress?

**Comment by subliun** [comment]
Groupchats are mostly done. I'm just waiting for toxcore to merge them at this point.

**Comment by HostFat** [comment]
Any news about this?

**Comment by dvinokurov-aw** [comment]
Any news? Groupchats are very important.

**Comment by dcharles525** [comment]
I believe the new groupchats is being tested in the core.

**Comment by SkyzohKey** [comment]
The last time I checked new group chats it was an issue with TCP. Dunno if any significant progress has been made but it would be nice to have current group chats while the new one are not ready.

**Comment by ProMcTagonist** [comment]
Why bother with that?

**Comment by zoff99** [comment]
any news?

**Comment by mystica555** [comment]
Ping? Anybody out there?

**Comment by wiiaam** [comment]
@mystica555 We are still waiting for the new groupchats to be merged into toxcore

**Comment by subliun** [comment]
And we shall wait forevermore.

**Comment by SkyzohKey** [comment]
> And we shall wait forevermore.

:+1:

**Comment by bo0ohXae** [comment]
I'm new to this, but it just makes no sense.. What does toxcore have to do with it, when every other client already supports group chats except the android one?

**Comment by ProMcTagonist** [comment]
Current clients have implemented current groupchats. These groupchats have many flaws and imperfections and are going to be replaced by New Groupchats in core pretty soon. Antox came along later than other clients, and before it implemented current groupchats, the developers decided just to wait for the new ones, because it would be a waste of time and effort to implement the groupchats that were going to be replaced soon anyway, and redo all their work.

**Comment by SkyzohKey** [comment]
@ProMcTagonist Ricin doesn't supports groupchats cuz we're waiting for the new groupchats. Honnestly supporting current group chat until the new ones gets merged in the master _(count at least 1 year)_ would takes me 4 little hours to fully-code and have it functionnal. I think it's not a question of time to code. But instead it is question of **how many time** it will take for the new groupchats getting merged ? At least an official post on the blog about a real timing to get them merged. This would permit new clients developpers to decide wether they have to add support for current groupchats or not.

Not that **I'm complaining**, here.

**Comment by ProMcTagonist** [comment]
@SkyzohKey I'm told that right now irungentoo is making dht work over tcp which is necessary for new groups to support tcp-only clients such as antox, eta uncertain

**Comment by SkyzohKey** [comment]
@ProMcTagonist Yep but why not merging the PR with TCP groups unavailable for now ? It would permit for clients to implement new groupchats working in UDP, falling back to the current groupchats if using TCP (during time TCP issue isn't resolved).

**Comment by ProMcTagonist** [comment]
> falling back to the current groupchats if using TCP

no, they're long gone

why turn one big break into two?

**Comment by SkyzohKey** [comment]
Or not supporting Old groupchats at all. But at least merging the new one. Even if their is not yet TCP support when the TCP issue will be close toxcore'll just need to be updates without causing any issue. Waiting more than 1 year for a such feature is freaking long. I guess we waited enough.

**Comment by zoff99** [comment]
waiting for godot ...

**Comment by LapFox** [comment]
@ProMcTagonist 

> ... replaced by New Groupchats in core _pretty soon_

Snails pace

**Comment by subliun** [comment]
>Snail's pace

That's disrespectful to snails

**Comment by je-vv** [comment]
Oh well, it seems as well there hadn't been any activity.  So is Antox abandoned then?

**Comment by marespiaut** [comment]
Hello,
Is there anything anyone can do to speed up the integration and availability of group chats?
I'm currently looking for a WhatsApp replacement, and group chat is a killer feature.

@subliun has mentioned that group chat were “mostly done” about _2.5 years ago_.

**Comment by ameenross** [comment]
@marespiaut I've been keeping an eye on Tox for a while, but have also tried many other chat clients/networks. Currently I recommend Kontalk.

**Comment by marespiaut** [comment]
@ameenross I'm using Kontalk with a friend.
It's OK so far, but it lacks a lot of features, and it seems (?) that the development of mobile and desktop clients is uneven.
Plus, there's no iOS client. I own no iOS devices, but my friends does, so that's a huge let down.

**Comment by marespiaut** [comment]
@ameenross You are right, but there's always compromise to be made if you want you application to be used by people. I can't blackmail my friends for using iOS.

**Comment by 2Belette** [comment]
I am wondering if there is anything which block the road for implementing it ? still core missing?

**Comment by SkyzohKey** [comment]
@2Belette yes. New group chats are vaporware and won't happens anytime soon. Do either wait for 5-6 years more (already 3 years since I said that they weren't going to be added soon) or implement the current one in Antic so this client actually start to be useful. In the meantime I recommend using TRiFA instead. It is more maintained that Antox.

**Comment by 2Belette** [comment]
Thanks! Yes it is what I realized regarding current group chats vs. potential future implementation...
And 3 years is a lot I agree.

I have tested TRiFA and to be honest the UI is quite old and I prefer much more Antox...
